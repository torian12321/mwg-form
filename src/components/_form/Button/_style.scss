@import '../../../styles/conf/index.scss';

// Vars
$btn-padding_v  : 8px !default;
$btn-padding_h  : 24px !default;


// Mixins
@mixin btn_palette(
  $color-main: $themeColor_1,
  $color-sec : smartColor($color-main, $white, $themeColor_1, 90)
) {
	color     : $color-sec;
	background: $color-main;
	border    : 1px $color-main solid;

	&:hover,
	&:focus {
		color       : $color-main;
		background  : $color-sec;
		border-color: $color-main;
	}
}

// Block
.btn{
  position       : relative;
  min-width      : 100px;
  padding        : $btn-padding_v $btn-padding_h;
  font-size      : 14px;
  cursor         : pointer;
  text-decoration: none;
	outline        : none !important;
	font-weight    : bold; 
  letter-spacing : 0.8px;

	@include rounded();
	@include shadow();
  @include transition($time:0.20s);

  // Status
  @include btn_palette();
  &:active,
  &.active{
    @include shadow()
  }
	&:disabled,
	&[disabled]{
		background  : $colorState_disable !important;
		border-color: $colorState_disable !important;
		color       : $fontColor_disable;
		box-shadow  : none !important;
		cursor      : not-allowed;
		@include opacity(0.8);
	}

  // Modifiers

	// Options
	&--large{
		min-width: 200px;
	}
	&--auto{
		min-width: auto;
	}

	// Styles
	&--main   { @include btn_palette(); }
	&--sec    { @include btn_palette($themeColor_2);}
  &--green  { @include btn_palette($green); }
	&--red    { @include btn_palette($red); }
	&--orange { @include btn_palette($orange); }
	&--white  { @include btn_palette($white); }

  &--black  { @include btn_palette($black); }
	&--grey   { @include btn_palette($grey); }
	&--blue   { @include btn_palette($blue); }
    
  &--info   { @include btn_palette($colorState_info); }
  &--success{ @include btn_palette($colorState_success); }
  &--warning{ @include btn_palette($colorState_warning); }
  &--danger { @include btn_palette($colorState_danger); }
}
