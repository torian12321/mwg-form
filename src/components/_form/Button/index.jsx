import React      from 'react';
import PropTypes  from 'prop-types';
import classNames from 'classnames';
import './_style.scss';

class Button extends React.Component {
  render() {
    const props = this.props;
    
    return (
      <button
        className= {classNames(
          'btn',
          props.color && `btn--${props.color}`,
          props.type  && `btn--${props.type}`,
          props.className
        )}
        onClick  = {props.onClick}
        disabled = {props.disabled}
      >
        {props.children}
      </button>
    )
  }
}


Button.propTypes = {
  children : PropTypes.any,
  className: PropTypes.string,
  disabled : PropTypes.bool,
  color    : PropTypes.oneOf(['main', 'sec', 'green', 'red', 'blue', 'white', 'grey', 'black']),
  type     : PropTypes.oneOf(['success', 'danger', 'warning']),
  onClick  : PropTypes.func
};
Button.defaultProps = {
  children : 'Submit',
  disabled : false
};

module.exports = Button;