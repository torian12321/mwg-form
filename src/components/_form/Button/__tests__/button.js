import React    from 'react';
import renderer from 'react-test-renderer';
import Button   from '../index';

describe('component', () => {
  describe('<Button />', () => {
    
    it('should render correctly', () => {
      const clickHander = jest.fn();
      const button = renderer.create(
        <Button onClick={clickHander} />
      ).toJSON();
      expect(button).toMatchSnapshot();

      // Click callback
      button.props.onClick();
      expect(clickHander).toBeCalled();
      expect(button.props.className).toEqual('btn');
      expect(button.children).toContain('Submit');
    });

    it('should have disabled', () => {
      const clickHander = jest.fn();
      const button = renderer.create(
        <Button disabled onClick={clickHander} />
      ).toJSON();

      button.props.onClick();
      expect(clickHander).toBeCalled();
      expect(button).toMatchSnapshot();
    });
    
  });
});
