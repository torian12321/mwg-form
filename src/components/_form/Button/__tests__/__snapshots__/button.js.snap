// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Button /> should have a large size 1`] = `
<button
  className="btn btn--threeD btn--large"
  disabled={false}
  onClick={undefined}
>
  Submit
</button>
`;

exports[`component <Button /> should have disabled 1`] = `
<button
  className="btn"
  disabled={true}
  onClick={[Function]}
>
  Submit
</button>
`;

exports[`component <Button /> should render correctly 1`] = `
<button
  className="btn"
  disabled={false}
  onClick={[Function]}
>
  Submit
</button>
`;
